invisible_cities/cities/penthesilea.py

```diff
@@ -85,28 +86,32 @@ def penthesilea(files_in, file_out, compression, event_range, print_mod, detecto
         write_hits            = df.sink(         hits_writer(h5out), args="hits")
         write_pointlike_event = df.sink(           kr_writer(h5out), args="pointlike_event")
         write_pmap_filter     = df.sink( event_filter_writer(h5out, "s12_selector"), args=("event_number", "pmap_passed"))
 
-        return push(source = pmap_from_files(files_in),
-                    pipe   = pipe(
-                        df.slice(*event_range, close_all=True),
-                        print_every(print_mod)                ,
-                        event_count_in       .spy             ,
-                        classify_peaks                        ,
-                        pmap_passed                           ,
-                        df.branch(write_pmap_filter)          ,
-                        pmap_select          .filter          ,
-                        event_count_out      .spy             ,
-                        df.fork((build_hits           , write_hits           ),
-                                (build_pointlike_event, write_pointlike_event),
-                                                        write_mc             ,
-                                                        write_event_info     )),
-                    result = dict(events_in  = event_count_in .future,
-                                  events_out = event_count_out.future,
-                                  selection  = pmap_select    .future))
+        event_ids = []
+        events_selected = df.sink(event_ids.append, args=("event_number"))
+
+        result = push(source = pmap_from_files(files_in),
+                      pipe   = pipe(
+                                    df.slice(*event_range, close_all=True),
+                                    print_every(print_mod)                ,
+                                    event_count_in       .spy             ,
+                                    classify_peaks                        ,
+                                    pmap_passed                           ,
+                                    df.branch(write_pmap_filter)          ,
+                                    pmap_select          .filter          ,
+                                    event_count_out      .spy             ,
+                                    df.fork((build_hits           , write_hits           ),
+                                            (build_pointlike_event, write_pointlike_event),
+                                                                    events_selected       ,
+                                                                    write_event_info     )),
+                      result = dict(events_in  = event_count_in .future,
+                                    events_out = event_count_out.future,
+                                    selection  = pmap_select    .future))
+
+        copy_mc_info(files_in, h5out, event_ids)
```

Comment:
I think it is safer and more verbose for someone trying to understand the code to make `copy_mc_info` run only when the run number is `<= 0` and not let it figure it out from the run number in the file

Reply:
Another side effect of calling `copy_mc_info` always is that the MC tables are created even for non-MC files!

Reply:
OK!

Reply:
Done.